elkodon_cal/src/event/mod.rs

```diff
@@ -14,26 +14,58 @@ pub enum NotifierNotifyError {
     InternalFailure,
 }
 
+impl std::fmt::Display for NotifierNotifyError {
```

Comment:
This was the only error from the concepts layer crate that was used in the examples. Shall I also implement the `Error` trait for the remaining errors in this crate

Reply:
@elBoberido I would postpone it for now and would explore `thiserror` and if we can somehow achieve `no_std` error implementation already in combination with rust nightly. And for rust stable we use `std::error::Error`.

Otherwise you may spend some time here and we refactor it again in the next weeks again.

Reply:
I found another solution in the zstd crate. They use `derive_more` with a feature flag to derive `Error` only on std. This makes would allow `no_std` on stable but without the `Error` trait. Something to think of.

They basically did this
```toml
[dependencies]
derive_more = { version = "0.99", default-features = false, features = ["display", "from"] }

[features]
default = ["std"]
std = ["derive_more/error"]
```

```rust
#[derive(Debug, derive_more::Display, derive_more::From)]
#[cfg_attr(feature = "std", derive(derive_more::Error))]
#[non_exhaustive]
pub enum Foo {
    #[display(fmt = "Bar occurred. Is: {baz}, must be either 1 or 2")]
    Bar { baz: u8 },
    //...
}
```

The attributes on the enum tags are quite similar to `thiserror` so it wouldn't be too hard to switch between the two crates.

Reply:
I created an issue #49 to track this.